.github/workflows/rust.yml

```diff
@@ -0,0 +1,49 @@
+on: push
+
+name: Continuous integration
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - 1.41.0  # MSRV
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+          components: rustfmt, clippy
+          target: thumbv7em-none-eabihf
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --target=thumbv7em-none-eabihf
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --examples --target=thumbv7em-none-eabihf
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: -- -D warnings
```

Comment:
Isn't this a bit much? Also, how about using the `clippy-check` action instead, since that'll put up inline annotations?

Reply:
I copied the action-rs example setup. Let me go learn about clippy-check. brb.

Reply:
How's that?

Reply:
Seems good!